google-services/Location/LocationUpdates/LocationUpdates/MainActivity.cs

```diff
@@ -1,227 +1,399 @@
+            mLastUpdateTimeTextView.Text = $"{mLastUpdateTimeLabel}: {mLastUpdateTime}";
+        }
+
+        public void StopLocationUpdates()
+        {
+            if (!mRequestingLocationUpdates)
+            {
+                Log.Debug(Tag, "stopLocationUpdates: updates never requested, no-op.");
+                return;
+            }
+            mFusedLocationClient.RemoveLocationUpdates(mLocationCallback).AddOnCompleteListener(this, new StopLocationUpdatesOnCompleteListener { Activity = this });
+        }
+
+        protected override void OnResume()
+        {
+            base.OnResume();
+            if (mRequestingLocationUpdates && CheckPermissions())
+            {
+                StartLocationUpdates();
+            }
+            else if (!CheckPermissions())
+            {
+                RequestPermissions();
+            }
+
+            UpdateUi();
+        }
+
+        protected override void OnPause()
+        {
+            base.OnPause();
+            StopLocationUpdates();
+        }
+
+        protected override void OnSaveInstanceState(Bundle savedInstanceState)
+        {
+            savedInstanceState.PutBoolean(KeyRequestingLocationUpdates, mRequestingLocationUpdates);
+            savedInstanceState.PutParcelable(KeyLocation, mCurrentLocation);
+            savedInstanceState.PutString(KeyLastUpdatedTimeString, mLastUpdateTime);
+            base.OnSaveInstanceState(savedInstanceState);
+        }
+
+        public void ShowSnackbar(int mainTextStringId, int actionStringId, View.IOnClickListener listener)
+        {
+            Snackbar.Make(FindViewById(Android.Resource.Id.Content), GetString(mainTextStringId), Snackbar.LengthIndefinite)
+                .SetAction(GetString(actionStringId), listener).Show();
+        }
+
+        public bool CheckPermissions()
+        {
+            var permissionState = ContextCompat.CheckSelfPermission(this, Manifest.Permission.AccessFineLocation);
+            return permissionState == (int)Permission.Granted;
+        }
+
+        public void RequestPermissions()
+        {
+            bool shouldProvideRationale = ActivityCompat.ShouldShowRequestPermissionRationale(this, Manifest.Permission.AccessFineLocation);
+
+            if (shouldProvideRationale)
+            {
+                Log.Info(Tag, "Displaying permission rationale to provide additional context.");
+                ShowSnackbar(Resource.String.permission_rationale, Android.Resource.String.Ok, new RequestPermissionsOnClickListener { Activity = this });
+            }
+            else
+            {
+                Log.Info(Tag, "Requesting permission");
+                ActivityCompat.RequestPermissions(this, new[] { Manifest.Permission.AccessFineLocation }, RequestPermissionsRequestCode);
+            }
+        }
+
+        public void OnRequestPermissionsResult(int requestCode, string[] permissions, int[] grantResults)
```

Comment:
Sorry I missed this earlier, however this should override the base method.

Reply:
Done.